### PR TITLE
add map view to best sub segments view

### PR DESCRIPTION
**Pull request by maglnet**
This PR does the following
* adds a map layer to the sub segments view
* highlights the segment when you move the mouse over the segment row
* changes the sub segments view size to "normal"
* on mobile devices (only tested on my phone) the path gets highlighted after you tap on the row

Open issues from reviews:
* [ ] multiple map support / main activity's map is destroyed when opening the sub segments view
* [ ] take care of activities without route information

I'm (as always) open to feedback :)

sub-segment-paths

### CODE REVIEW

**Comment by maglnet** [comment]
I just realized that I currently don't care about the configured maps step size, when creating the paths. Where and how should I get the step size?

**Comment by laufhannes** [comment]
Great to see that, thanks!

The step size is available via the old configuration (deprecated and only possible if the old Frontend class is used in that controller):
```
(int)\Runalyze\Configuration::ActivityView()->routePrecision()->value()
```

or in the controller:
```
$this->get('app.configuration_manager')->getList()->getActivityView()->get('GMAP_PATH_PRECISION');
```
Feel free to add a `getMapPrecision() : int` to the activity view category.

or in templates:
```
{{ configVar('activity-view.GMAP_PATH_PRECISION') }}
```

**Comment by maglnet** [comment]
Ok, I'll adjust this tomorrow :)

**Comment by maglnet** [comment]
I adjusted the paths to take care of the precision parameter.  
I did not understand your hint where to add the `getMapPrecision() : int`. If you could clarify this I could go on adjusting this.

**Comment by laufhannes** [comment]
It's just replacing the last call `get('GMAP_PATH_PRECISION')` (and int cast) by a new method `getMapPrecision()` in [ActivityView](https://github.com/Runalyze/Runalyze/blob/master/src/CoreBundle/Component/Configuration/Category/ActivityView.php) such that the controller does not need to know the exact key and its database type.

File: src/CoreBundle/Controller/ActivityController.php [hunk 1]
```diff
@@ -405,10 +406,27 @@ public function subSegmentInfoAction($id, Account $account, Trackdata $trackdata
         $statistics->setTimesToAnalyze([30, 60, 120, 300, 600, 720, 1800, 3600, 7200]);
         $statistics->findSegments();
 
+        $Frontend = new \Frontend(false, $this->get('security.token_storage'));
+        $context = new Context($id, $account->getId());
+        $route = $context->route();
```
Comment (laufhannes) [inline src/CoreBundle/Controller/ActivityController.php]: What I did not think about so far: I guess we need to catch activities without route (or with route but without geohashes), e.g. treadmill runs, which do have a distance array but not necessarily gps data.

**Comment by laufhannes** [comment]
There's another major issue: Opening the best segment analysis removes the original activity view's map. Our current leaflet wrapper does support only one map at the same time. We probably need to make `object` and `control.layers` arrays in `RunalyzeLeaflet` and fix all calls to always send the current id as one argument.

Let me know if you want me to take care of that.

**Comment by maglnet** [comment]
Well, I could check the problem with multiple maps in the next days, too, but if you would like to adjust it, feel free :-)

I'll add the issues to the main description so we don't forget anything.

**Comment by maglnet** [comment]
@laufhannes I was thinking of implementing a stateful jquery-ui widget for the maps but this would result in a bc break, how do you think about this?

reference: https://learn.jquery.com/plugins/stateful-plugins-with-widget-factory/

**Comment by laufhannes** [comment]
Fixed by #2302.

**Pull request closed without merging**